### PR TITLE
Conversion PearDatabase to yii Leads_Module_Model

### PR DESCRIPTION
**Pull request by mlorencik**

### CODE REVIEW

File: modules/Leads/models/Module.php [hunk 1]
```diff
@@ -14,35 +14,33 @@ class Leads_Module_Model extends Vtiger_Module_Model
...
 	public function getLeadsCreated($owner, $dateFilter)
 	{
-		$db = PearDatabase::getInstance();
-		$module = $this->getName();
-		$securityParameter = \App\PrivilegeQuery::getAccessConditions($module);
+		$query = (new App\Db\Query())->select(['count' => 'COUNT(*)', 'time' => 'week(createdtime)'])
+			->from('vtiger_leaddetails')
+			->innerJoin('vtiger_crmentity', ['vtiger_leaddetails.leadid' => 'vtiger_crmentity.crmid'])
+			->where(['deleted' => 0, 'converted' => 0]);
+		$securityParameter = \App\PrivilegeQuery::getAccessConditions($this->getName());
```
Comment (TomaszQr) [inline modules/Leads/models/Module.php]: App\PrivilegeQuery::getConditions(...)

File: modules/Leads/models/Module.php [hunk 2]
```diff
@@ -14,35 +14,33 @@ class Leads_Module_Model extends Vtiger_Module_Model
...
+			$query->andWhere(['AND', ['BETWEEN', 'createdtime', $dateFilter['start'] . ' 00:00:00', $dateFilter['end'] . ' 23:59:59']]);
 		}
-
-		$sql = sprintf('SELECT COUNT(*) AS count, date(createdtime) AS time FROM vtiger_leaddetails
-		INNER JOIN vtiger_crmentity ON vtiger_leaddetails.leadid = vtiger_crmentity.crmid
-		WHERE deleted = 0 %s %s %s', $ownerSql, $dateFilterSql, $securityParameter);
-		$sql .= ' && converted = 0 GROUP BY week(createdtime)';
-		$result = $db->pquery($sql, $params);
+		$query->groupBy('week(createdtime)');
+		$dataReader = $query->createCommand()
```
Comment (TomaszQr) [inline modules/Leads/models/Module.php]: $dataReader = $query->groupBy('week(createdtime)')->createCommand() ...........

File: modules/Leads/models/Module.php [hunk 3]
```diff
@@ -14,35 +14,33 @@ class Leads_Module_Model extends Vtiger_Module_Model
...
+			$query->andWhere(['smownerid' => $owner]);
 		}
-
-		$params = [];
 		if (!empty($dateFilter)) {
-			$dateFilterSql = ' && createdtime BETWEEN ? AND ? ';
-			//client is not giving time frame so we are appending it
-			$params[] = $dateFilter['start'] . ' 00:00:00';
-			$params[] = $dateFilter['end'] . ' 23:59:59';
+			$query->andWhere(['AND', ['BETWEEN', 'createdtime', $dateFilter['start'] . ' 00:00:00', $dateFilter['end'] . ' 23:59:59']]);
```
Comment (TomaszQr) [inline modules/Leads/models/Module.php]: ?

File: modules/Leads/models/Module.php [hunk 4]
```diff
@@ -14,35 +14,30 @@ class Leads_Module_Model extends Vtiger_Module_Model
...
+	 * @param array $dateFilter
 	 *
 	 * @return <Array>
 	 */
 	public function getLeadsCreated($owner, $dateFilter)
 	{
-		$db = PearDatabase::getInstance();
-		$module = $this->getName();
-		$securityParameter = \App\PrivilegeQuery::getAccessConditions($module);
+		$query = (new App\Db\Query())->select(['count' => 'COUNT(*)', 'time' => 'date(createdtime)'])
```
Comment (mariuszkrzaczkowski) [inline modules/Leads/models/Module.php]: date(createdtime)  - SQL 89 or 92 ?

File: modules/Leads/models/Module.php [hunk 5]
```diff
@@ -14,35 +14,30 @@ class Leads_Module_Model extends Vtiger_Module_Model
...
-			$params[] = $dateFilter['end'] . ' 23:59:59';
+			$query->andWhere(['between', 'createdtime', $dateFilter['start'] . ' 00:00:00', $dateFilter['end'] . ' 23:59:59']);
 		}
-
-		$sql = sprintf('SELECT COUNT(*) AS count, date(createdtime) AS time FROM vtiger_leaddetails
-		INNER JOIN vtiger_crmentity ON vtiger_leaddetails.leadid = vtiger_crmentity.crmid
-		WHERE deleted = 0 %s %s %s', $ownerSql, $dateFilterSql, $securityParameter);
-		$sql .= ' && converted = 0 GROUP BY week(createdtime)';
-		$result = $db->pquery($sql, $params);
+		$dataReader = $query->groupBy('date(createdtime)')
```
Comment (mariuszkrzaczkowski) [inline modules/Leads/models/Module.php]: date(createdtime) - SQL 89 or 92 ?

Comment (mlorencik) [reply modules/Leads/models/Module.php]: https://en.wikipedia.org/wiki/SQL-92

**Pull request merged**